src/media/local-stream.ts

```diff
@@ -283,6 +283,36 @@ abstract class _LocalStream extends Stream {
     return this.effects;
   }
 
+  /**
+   * Method to serialize data about input, output streams
+   * and also effects from LocalStream.
+   *
+   * @returns - A JSON-compatible object representation with data from LocalStream.
+   */
+  toJSON() {
+    return {
+      inputStream: {
+        active: this.inputStream.active,
+        id: this.inputStream.id,
+        muted: this.muted,
+        label: this.label,
+        enabled: this.inputTrack.enabled,
```

Comment:
@bbaldino @brycetham not sure that for one case here it is useful to have a getter for `enabled`. So right now put it via `inputTrack`. Let me know if you have any other thoughts

Reply:
Yeah I wouldn't add a getter for `enabled` unless there is a need for it outside this function. Actually, I would argue that we shouldn't include `enabled` here at all, since `muted` already takes `enabled` into account.

I would also probably move `muted`, `label`, and `readyState` out of `inputStream` and put it at the same level as `inputStream`, `outputStream`, and `effects`. These things, in my opinion, apply to the whole `LocalStream`, not just the input (hence why we have explicitly defined getters for all of them).

Reply:
for debugging purposes it's better to keep both `muted` and `enabled`

Reply:
Ok, but let's keep it separate from the overall `LocalStream` mute state. So I would suggest something like this:

```
{
  muted: this.muted,
  inputStream: {
    muted: this.inputTrack.muted,
    enabled: this.inputTrack.enabled,
    ...
  },
  ...
}
```

Reply:
@brycetham updated recording your advice, thanks
@bbaldino @marcin-bazyl FYI